### PR TITLE
[SPG-508]: adding service overrides TF resource

### PR DESCRIPTION
**Pull request by RajBaviskar**
* adding service overrides TF resource

### CODE REVIEW

File: helpers/schema.go [hunk 1]
```diff
@@ -127,6 +127,11 @@ func SetProjectLevelResourceSchema(s map[string]*schema.Schema) {
 	s["project_id"] = GetProjectIdSchema(SchemaFlagTypes.Required)
 }
 
+func SetProjectLevelResourceSchemaForServiceOverride(s map[string]*schema.Schema) {
```
Comment (rathodmeetsatish) [inline helpers/schema.go:130]: this is service override specific please move it to service override resource

File: internal/service/platform/environment_service_overrides/resource_environment_service_overrides.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+func readEnvironmentServiceOverridesList(d *schema.ResourceData, env *nextgen.PageResponseServiceOverrideResponse) {
+	ServiceOverrideList := env.Content
+	for _, value := range ServiceOverrideList {
+		readEnvironmentServiceOverrides(d, &value)
+	}
+}
+
+func readEnvironmentServiceOverrides(d *schema.ResourceData, so *nextgen.ServiceOverrideResponse) {
+	d.SetId(so.OrgIdentifier) //temp id
```
Comment (rathodmeetsatish) [inline internal/service/platform/environment_service_overrides/resource_environment_service_overrides.go:139]: @yogesh-chauhan can you please the id for get call that we discussed was of format "service_id+env_id"

File: helpers/schema.go (hunk 1 above)
Comment (RajBaviskar) [reply helpers/schema.go]: changed.

File: internal/service/platform/environment_service_overrides/resource_environment_service_overrides.go (hunk 2 above)
Comment (RajBaviskar) [reply internal/service/platform/environment_service_overrides/resource_environment_service_overrides.go:139]: we have this limitation  "Error: identifier: can be 64 characters long and can only contain alphanumeric, underscore and $ characters, and not start with a number" 

I think we can go with underscore _

File: examples/resources/harness_environment_service_overrides/resource.tf [hunk 3]
```diff
@@ -0,0 +1,41 @@
+resource "harness_environment_service_overrides" "test" {
+  identifier = "%[1]s"
```
Comment (rathodmeetsatish) [inline examples/resources/harness_environment_service_overrides/resource.tf:2]: identifier = "identifier"

File: examples/resources/harness_environment_service_overrides/resource.tf [hunk 4]
```diff
@@ -0,0 +1,41 @@
+resource "harness_environment_service_overrides" "test" {
```
Comment (manavjot-harness) [inline examples/resources/harness_environment_service_overrides/resource.tf:1]: shouldn't this be "harness_environment_service_overrides" "example" ?
and the fields have dummy values in the example directory?

Comment (RajBaviskar) [reply examples/resources/harness_environment_service_overrides/resource.tf:1]: fixed.

File: examples/resources/harness_environment_service_overrides/resource.tf (hunk 3 above)
Comment (RajBaviskar) [reply examples/resources/harness_environment_service_overrides/resource.tf]: changed.

File: examples/data-sources/harness_environment_service_overrides/data-source.tf [hunk 5]
```diff
@@ -0,0 +1,7 @@
+data "harness_environment_service_overrides" "test" {
+  identifier = harness_environment_service_overrides.test.id
```
[the remark this answers is not in the document]
Comment (RajBaviskar) [reply examples/data-sources/harness_environment_service_overrides/data-source.tf]: changed

File: docs/resources/platform_environment_service_overrides.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+Resource for creating a Harness environment service overrides.
+
+
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `env_id` (String) The env ID to which the overrides associated.
```
Comment (rashminandasahoo) [inline docs/resources/platform_environment_service_overrides.md:20]: Refer to the previous comments.

File: docs/resources/platform_environment_service_overrides.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+- `identifier` (String) identifier of the service overrides.
```
Comment (rashminandasahoo) [inline docs/resources/platform_environment_service_overrides.md:21]: Identifier of the service override.

**Pull request merged**